projects/rocprofiler-systems/source/lib/core/agent_info.hpp

```diff
@@ -0,0 +1,237 @@
+    data["capability"]["WaveLaunchModeSupported"] =
+        agent_data.capability.ui32.WaveLaunchModeSupported;
+    data["capability"]["PreciseMemoryOperationsSupported"] =
+        agent_data.capability.ui32.PreciseMemoryOperationsSupported;
+    data["capability"]["DEPRECATED_SRAM_EDCSupport"] =
+        agent_data.capability.ui32.DEPRECATED_SRAM_EDCSupport;
+    data["capability"]["Mem_EDCSupport"]  = agent_data.capability.ui32.Mem_EDCSupport;
+    data["capability"]["RASEventNotify"]  = agent_data.capability.ui32.RASEventNotify;
+    data["capability"]["ASICRevision"]    = agent_data.capability.ui32.ASICRevision;
+    data["capability"]["SRAM_EDCSupport"] = agent_data.capability.ui32.SRAM_EDCSupport;
+    data["capability"]["SVMAPISupported"] = agent_data.capability.ui32.SVMAPISupported;
+    data["capability"]["CoherentHostAccess"] =
+        agent_data.capability.ui32.CoherentHostAccess;
+    data["capability"]["DebugSupportedFirmware"] =
+        agent_data.capability.ui32.DebugSupportedFirmware;
+
+    data["cu_per_engine"]      = agent_data.cu_per_engine;
+    data["max_waves_per_cu"]   = agent_data.max_waves_per_cu;
+    data["family_id"]          = agent_data.family_id;
+    data["workgroup_max_size"] = agent_data.workgroup_max_size;
+    data["grid_max_size"]      = agent_data.grid_max_size;
+    data["local_mem_size"]     = agent_data.local_mem_size;
+    data["hive_id"]            = agent_data.hive_id;
+    data["gpu_id"]             = agent_data.gpu_id;
+
+    data["workgroup_max_dim"]["x"] = agent_data.workgroup_max_dim.x;
+    data["workgroup_max_dim"]["y"] = agent_data.workgroup_max_dim.y;
+    data["workgroup_max_dim"]["z"] = agent_data.workgroup_max_dim.z;
+
+    data["grid_max_dim"]["x"] = agent_data.grid_max_dim.x;
+    data["grid_max_dim"]["y"] = agent_data.grid_max_dim.y;
+    data["grid_max_dim"]["z"] = agent_data.grid_max_dim.z;
+
+    data["name"] = agent_data.name ? std::string(agent_data.name) : "";
+    data["vendor_name"] =
+        agent_data.vendor_name ? std::string(agent_data.vendor_name) : "";
+    data["product_name"] =
+        agent_data.product_name ? std::string(agent_data.product_name) : "";
+    data["model_name"] = agent_data.model_name ? std::string(agent_data.model_name) : "";
+
+    data["node_id"]              = agent_data.node_id;
+    data["logical_node_id"]      = agent_data.logical_node_id;
+    data["logical_node_type_id"] = agent_data.logical_node_type_id;
+
+    data["runtime_visibility"]["hsa"]       = agent_data.runtime_visibility.hsa;
+    data["runtime_visibility"]["hip"]       = agent_data.runtime_visibility.hip;
+    data["runtime_visibility"]["rccl"]      = agent_data.runtime_visibility.rccl;
+    data["runtime_visibility"]["rocdecode"] = agent_data.runtime_visibility.rocdecode;
+
+    auto& uuid_bytes = data["uuid"]["bytes"];
+    for(size_t i = 0; i < 16; ++i)
+    {
+        uuid_bytes[std::string("value") + std::to_string(i)] = agent_data.uuid.bytes[i];
+    }
```

Comment:
The pre-processor check is required because of a structure change between 0.6.0 and 1.0.0.

Do you think that this routine is easier to understand? 

```suggestion
    #if(ROCPROFILER_VERSION >= 700)
    data["uuid"]["bytes"] = nlohmann::json::array();
    for(size_t i = 0; i < 16; ++i)
    {
        data["uuid"]["bytes"].push_back(agent_data.uuid.bytes[i]);
    }
    #endif
```

Or possibly something like this, if we just want to store the UUID as a formatted string, ex UUID: 00ff7480-0000-1000-8000-000000000000

```suggestion
    #if(ROCPROFILER_VERSION >= 700)
  
    // Convert UUID bytes to standard hex string format
    std::stringstream uuid_ss;
    uuid_ss << std::hex << std::setfill('0');
    
    for(size_t i = 0; i < 16; ++i)
    {
        uuid_ss << std::setw(2) << static_cast<unsigned>(agent_data.uuid.bytes[i]);
        
        // Add hyphens at standard UUID positions: 8-4-4-4-12
        if(i == 3 || i == 5 || i == 7 || i == 9)
            uuid_ss << '-';
    }
    
    data["uuid"] = uuid_ss.str();
    #endif
```